alacritty/src/renderer/mod.rs

```diff
@@ -218,3 +220,38 @@ impl Renderer {
         }
     }
 }
+
+struct GlExtensions;
+
+impl GlExtensions {
+    /// Check if the given `extension` is supported.
+    ///
+    /// This function will lazyly load OpenGL extensions.
+    fn contains(extension: &str) -> bool {
+        static OPENGL_EXTENSIONS: OnceCell<HashSet<&'static str>> = OnceCell::new();
+
+        OPENGL_EXTENSIONS
+            .get_or_init(|| unsafe {
+                let extensions = gl::GetString(gl::EXTENSIONS);
```

Comment:
Moving this entire closure into a separate function would probably clean things up a lot visually.

Reply:
Decided to just do this myself real quick.